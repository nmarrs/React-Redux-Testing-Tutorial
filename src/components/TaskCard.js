import React from 'react'
import PropTypes from 'prop-types'

import './styles/TaskCard.css'

const TaskCard = (props) => {
  const {
    task,
    toggleEditTaskModal
  } = props

  return (
    <div className="card shadow font-title mt-3 TaskCard-card mx-auto">
      {/* eslint-disable-next-line */}
      <a onClick={() => toggleEditTaskModal(true, task)}>
        <div className='card-title mt-3'>
          <h5>{task.title}</h5>
        </div>
        <div className='card-body'>
          <table className='table table-hover'>
            <tbody>
              <tr>
                <th><i className="fas fa-exclamation"></i> Priority Level (1-5)</th>
                <td>{task.priority}</td>
              </tr>
              <tr>
                <th><i className="far fa-clock"></i> Estimate (hours)</th>
                <td>{task.timeEstimate}</td>
              </tr>
            </tbody>
          </table>
        </div>
      </a>
    </div>
  )
}

TaskCard.propTypes = {
  task: PropTypes.object.isRequired,
  toggleEditTaskModal: PropTypes.func.isRequired
}

export default TaskCard
