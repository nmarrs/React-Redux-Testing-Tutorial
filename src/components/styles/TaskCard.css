.TaskColumn-card {
  max-width: 300px;
  text-align: center;
}
