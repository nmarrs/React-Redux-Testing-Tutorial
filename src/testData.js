export let testTask = {
  id: 1,
  title: 'Unload merchandise',
  priority: 3,
  timeEstimate: 2,
  notes: 'Unload merchandise from the bay door. Driver usually comes by at 6 am sharp.',
  feedback: '',
  startTime: '',
  endTime: '',
  completionTime: '',
  currentStatus: 'To Do',
  rankWeight: 150
}

export let testTask1 = {
  id: 2,
  title: 'Stock shelves with unpacked items',
  priority: 1,
  timeEstimate: 10,
  notes: 'Take merchandise items and stock shelves with them. Be sure to place items in correct areas.',
  feedback: '',
  startTime: '',
  endTime: '',
  completionTime: '',
  currentStatus: 'To Do',
  rankWeight: 10
}

export let testTask2 = {
  id: 3,
  title: 'Stock shelves with unpacked items',
  priority: 5,
  timeEstimate: 2,
  notes: 'Take merchandise items and stock shelves with them. Be sure to place items in correct areas.',
  feedback: '',
  startTime: '',
  endTime: '',
  completionTime: '',
  currentStatus: 'To Do',
  rankWeight: 250
}
