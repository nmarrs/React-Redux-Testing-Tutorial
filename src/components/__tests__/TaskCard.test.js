import React from 'react'
import { shallow } from 'enzyme'

import TaskCard from '.././TaskCard'
import { testTask } from '../../testData'

describe('TaskCard', () => {
  let wrapper
  let toggleEditTaskModal = jest.fn()

  beforeEach(() => {
    wrapper = shallow(<TaskCard
      task={testTask}
      toggleEditTaskModal={toggleEditTaskModal} />)
  })

  it('should exist', () => {
    expect(wrapper).toBeDefined()
  })

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('taskCard element handles click event as expected', () => {
    const taskCardElement = wrapper.find('a')

    taskCardElement.simulate('click')

    expect(toggleEditTaskModal).toHaveBeenCalled()
    expect(toggleEditTaskModal).toHaveBeenCalledWith(true, testTask)
  })
})
