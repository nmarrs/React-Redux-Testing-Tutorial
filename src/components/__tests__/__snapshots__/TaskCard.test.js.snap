// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskCard renders correctly 1`] = `
<div
  className="card shadow font-title mt-3 TaskCard-card mx-auto"
>
  <a
    onClick={[Function]}
  >
    <div
      className="card-title mt-3"
    >
      <h5>
        Unload merchandise
      </h5>
    </div>
    <div
      className="card-body"
    >
      <table
        className="table table-hover"
      >
        <tbody>
          <tr>
            <th>
              <i
                className="fas fa-exclamation"
              />
               Priority Level (1-5)
            </th>
            <td>
              3
            </td>
          </tr>
          <tr>
            <th>
              <i
                className="far fa-clock"
              />
               Estimate (hours)
            </th>
            <td>
              2
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </a>
</div>
`;
